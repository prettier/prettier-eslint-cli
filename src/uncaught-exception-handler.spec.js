import getLoggerMock from 'loglevel-colored-level-prefix';

import onUncaughtException from './uncaught-exception-handler';

jest.mock('loglevel-colored-level-prefix', () => {
  const logger = {};
  const __mock__ = { logger, level: 4, resetAll };
  const getLogger = jest.fn(() => resetAll());
  getLogger.__mock__ = __mock__;
  return getLogger;

  function resetAll() {
    getLogger.mockClear();
    Object.assign(logger, {
      getLevel: jest.fn(() => getLogger.__mock__.level),
      trace: jest.fn(),
      debug: jest.fn(),
      info: jest.fn(),
      warn: jest.fn(),
      error: jest.fn(),
    });
    return logger;
  }
});

beforeEach(() => {
  getLoggerMock.__mock__.resetAll();
});

test('logs all options', () => {
  const logger = getLoggerMock();
  runWithCatch(new Error('my error'));
  expect(logger.error).toHaveBeenCalledTimes(1);
  const errorLog = logger.error.mock.calls[0].join(' ');
  expect(errorLog).toMatchSnapshot();
});

test('logs a check for trace', () => {
  getLoggerMock.__mock__.level = 0;
  const logger = getLoggerMock();
  runWithCatch(new Error('my error'));
  expect(logger.error).toHaveBeenCalledTimes(1);
  const errorLog = logger.error.mock.calls[0].join(' ');
  expect(errorLog).toContain('âœ…');
  expect(errorLog).toMatchSnapshot();
});

test('re-throws the given error', () => {
  const myError = new Error('my error');
  expect(() => onUncaughtException(myError)).toThrow(myError);
});

function runWithCatch(...args) {
  try {
    onUncaughtException(...args);
  } catch {
    // ignore
  }
}
