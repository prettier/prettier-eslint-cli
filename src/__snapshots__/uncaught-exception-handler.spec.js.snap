// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs a check for trace 1`] = `
"There has been an unknown error when running the prettier-eslint CLI. If it's unclear to you what went wrong, then try this: 
  âœ…  Run the script again with the LOG_LEVEL environment variable set to "trace"
  2. Search existing issues on GitHub: https://github.com/prettier/prettier-eslint-cli/issues?utf8=%E2%9C%93&q=my%20error
  3. Make a minimal reproduction in a totally separate repository. You can fork this one: https://github.com/kentcdodds/prettier-eslint-cli-repro
  4. Post an issue with a link to your reproduction to the issues on GitHub: https://github.com/prettier/prettier-eslint-cli/issues/new"
`;

exports[`logs all options 1`] = `
"There has been an unknown error when running the prettier-eslint CLI. If it's unclear to you what went wrong, then try this: 
  1. Run the script again with the LOG_LEVEL environment variable set to "trace"
  2. Search existing issues on GitHub: https://github.com/prettier/prettier-eslint-cli/issues?utf8=%E2%9C%93&q=my%20error
  3. Make a minimal reproduction in a totally separate repository. You can fork this one: https://github.com/kentcdodds/prettier-eslint-cli-repro
  4. Post an issue with a link to your reproduction to the issues on GitHub: https://github.com/prettier/prettier-eslint-cli/issues/new"
`;
